### DISCUSSION TITLE
Install and Build Cache

### DISCUSSION THREAD

**Discussion by vikeshmittal** [code-shaper/code-shaper#9] [General]
Hi,

This seems like a great code generator. I wanted to ask if this also provides any optimization for install and build for CI/CD, for example caching the previous pipeline. I have seen this as a big problem with most of mono-repos that the overall developer experience starts decreasing as the code size increases over time.

Thanks,
Vikesh M

**Comment by nareshbhatia** [comment]
Hi @vikeshmittal, glad you liked our work.

To answer your question, optimization of install & build for CI/CD is not part of the Code Shaper scope. Our basic focus is to provide a code generation engine that can be used to build any tech stack including a CI/CD pipeline on top of a monorepo. The optimization of the CI/CD pipeline would be specific to the underlying monorepo technology and in the scope of the that stack.

**Reply by vikeshmittal** [reply]
Got it, thanks. I got a bit confused by title ‘Create an empty monorepo’ in the documentation and thought it is backed by tools like Rush, Nx or Bazel.
